src/sst/elements/vanadis/inst/vmulhigh.h

```diff
@@ -0,0 +1,88 @@
+// Copyright 2009-2022 NTESS. Under the terms
+// of Contract DE-NA0003525 with NTESS, the U.S.
+// Government retains certain rights in this software.
+//
+// Copyright (c) 2009-2022, NTESS
+// All rights reserved.
+//
+// Portions are copyright of other developers:
+// See the file CONTRIBUTORS.TXT in the top level directory
+// of the distribution for more information.
+//
+// This file is part of the SST software package. For license
+// information, see the LICENSE file in the top level directory of the
+// distribution.
+
+#ifndef _H_VANADIS_MUL_HIGH
+#define _H_VANADIS_MUL_HIGH
+
+#include "inst/vinst.h"
+
+namespace SST {
+namespace Vanadis {
+
+template<typename resultType, typename left_format, typename right_format, typename tmpType, bool signed_left, bool signed_right>
+class VanadisMultiplyHighInstruction : public VanadisInstruction
+{
+public:
+    VanadisMultiplyHighInstruction(
+        const uint64_t addr, const uint32_t hw_thr, const VanadisDecoderOptions* isa_opts, const uint16_t dest,
+        const uint16_t src_1, const uint16_t src_2) :
+        VanadisInstruction(addr, hw_thr, isa_opts, 2, 1, 2, 1, 0, 0, 0, 0)
+    {
+
+        isa_int_regs_in[0]  = src_1;
+        isa_int_regs_in[1]  = src_2;
+        isa_int_regs_out[0] = dest;
+    }
+
+    VanadisMultiplyHighInstruction* clone() override { return new VanadisMultiplyHighInstruction(*this); }
+
+    VanadisFunctionalUnitType getInstFuncType() const override { return INST_INT_ARITH; }
+    const char*               getInstCode() const override
+    {
+        if( signed_left && signed_right) {
```

Comment:
Can you set these so they are `MULH32` etc? In RISCV I think `MULH`, `MULHU` and `MULHSU` are all 32bit width results (even though the multiplication is performed by a 64-bit multiply to extract the higher 32 bits.

Reply:
The same comment here. XLEN is 64, so it should produce a 128b result and put the upper 64b in the destination register.